vignettes/customize.Rmd

```diff
@@ -7,40 +7,257 @@ vignette: >
+instead of the one associated with the parent class.
+
+To illustrate this we can take a look at the `list_content_cnt()` generic:
+
+```{r}
+# Print the generic 
+print(list_content_cnt)
+
+# List the registered s3 methods
+methods("list_content_cnt") |> 
+  cat(sep = "\n")
+```
+Building further on the example below we can define a custom method for `list_content_cnt()` for the `connector_project` class,
+to be used instead of `list_content_cnt.connector_fs()`:
+
+```{r}
+list_content_cnt.connector_project <- function(connector_object, ...) {
+  cli::cli_alert("Listing content of {connector_object$project}")
+  NextMethod()
+}
+```
+
+This is of course a very simple example, that just prints a message before calling the `connector_fs` method.
+
+We can now see that this method is available and that is it associated with the `connector_project` class:
+
+```{r}
+# List methods again
+methods("list_content_cnt") |> 
+  cat(sep = "\n")
+
+# Print my_project connector to see associated methods
+print(my_project)
+```
+And when we use `list_content_cnt()` on our `my_project` object, we see that the custom method is used and we get the message:
+
+```{r}
+my_project |> 
+  list_content_cnt()
+```
+
+## Use extra class for simple customization
+
+If you as above just want to slightly tweak the behavior of an existing functionality
+an alternative solution is to use the `extra_class` argument when initializing of the connector.
+
+This argument adds the `extra_class` as the first class of the creating the connector,
+meaning that for any generic dispatch, such as of `list_content_cnt()`, a method for this class will
+be used before any of the connector classes.
+
+To redo the two examples above we make a new `connector_fs` with the extra class `demo_extra`:
+
+```{r}
+my_project_extra <- connector_fs$new(
+  path = "my_root_path/my_project", 
+  extra_class = "my_extra_class"
+  )
+
+print(my_project_extra)
+```
+As you can see here we have all the methods from `connector_fs()`, but the `demo_extra` is now the first class in the class hierarchy.
+
+To create a custom method for `list_content_cnt()` for the `demo_extra` we do the same as for the `connector_project` above:
+
+```{r}
+list_content_cnt.my_extra_class <- function(connector_object, ...) {
+  cli::cli_alert("Listing content of {connector_object$path}")
+  NextMethod()
+}
+```
+
+The project information is of course not available now, so we just print the path instead,
+but otherwise everything is the same:
+
+```{r}
+# List methods
+methods("list_content_cnt")
+
+# Print my_project_extra connector to see associated methods
+print(my_project_extra)
+
+# List content to see the new message
+my_project_extra |> 
+  list_content_cnt()
 ```
 
+## Special handling of files
+
+A special property of file storage connectors (inheriting from `connector_fs()`) is
+that they are operating on files on not databases. This means that they can handle 
```

Comment:
small typo no : "on not databases" and may be or just a coma ?

Reply:
Think I just got them in the wrong order. Does "not on databases" work?

Reply:
yes :)